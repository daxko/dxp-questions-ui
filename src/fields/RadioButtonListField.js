var React = require('react');

var RadioButtonListField = React.createClass({

	onChange: function(e) {
		this.props.onChange(e);
		this.props.onBlur();
	},

	render: function() {

		var question = this.props.question;

		return (
			<div className="dxp-radio">
				{ 
					question.possible_answers.map(function(pa) {
						return (
							<label key={pa.id}>
								<input type="radio" 
									name={this.props.question_id} 
									value={pa.id}
									disabled={this.props.question.read_only}
									onChange={this.onChange}
									onBlur={this.props.onBlur}
									checked={pa.id == this.props.answer || this.props.answer == null && pa.id == this.props.question.default_value} 
									className={this.props.className} />
								{pa.display_value}
							</label>
						);
					}.bind(this))
				}
			</div>
		);
	}

});

module.exports = RadioButtonListField;