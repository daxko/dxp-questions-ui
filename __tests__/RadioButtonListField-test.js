import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import RadioButtonListField from '../src/fields/RadioButtonListField'
const json = require('../snippets/json/radio_question.json')

configure({adapter: new Adapter() })

describe('RadioButtonListField component', () => {
  it('should render a question', () => {
    const wrapper = shallow(<RadioButtonListField question={json} />)
    expect(wrapper).toMatchSnapshot()
  })
})
