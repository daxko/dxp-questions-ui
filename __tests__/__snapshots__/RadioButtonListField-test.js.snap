// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonListField component should render a question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioButtonListField
    question={
      Object {
        "default_value": null,
        "description": "",
        "empty_value": "",
        "hide_empty_option": false,
        "possible_answers": Array [
          Object {
            "display_value": "Yes",
            "id": "childcare_possible_answer-103473",
            "value": "Yes",
          },
          Object {
            "display_value": "No",
            "id": "childcare_possible_answer-103474",
            "value": "No",
          },
        ],
        "read_only": false,
        "required": false,
        "title": "Have you participated in this program before?",
        "type": "radio",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          <input
            checked={false}
            className={undefined}
            disabled={false}
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            type="radio"
            value="childcare_possible_answer-103473"
          />
          Yes
        </label>,
        <label>
          <input
            checked={false}
            className={undefined}
            disabled={false}
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            type="radio"
            value="childcare_possible_answer-103474"
          />
          No
        </label>,
      ],
      "className": "dxp-radio",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "childcare_possible_answer-103473",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              className={undefined}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="radio"
              value="childcare_possible_answer-103473"
            />,
            "Yes",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "className": undefined,
              "disabled": false,
              "name": undefined,
              "onBlur": undefined,
              "onChange": [Function],
              "type": "radio",
              "value": "childcare_possible_answer-103473",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Yes",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "childcare_possible_answer-103474",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              className={undefined}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="radio"
              value="childcare_possible_answer-103474"
            />,
            "No",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "className": undefined,
              "disabled": false,
              "name": undefined,
              "onBlur": undefined,
              "onChange": [Function],
              "type": "radio",
              "value": "childcare_possible_answer-103474",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "No",
        ],
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            <input
              checked={false}
              className={undefined}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="radio"
              value="childcare_possible_answer-103473"
            />
            Yes
          </label>,
          <label>
            <input
              checked={false}
              className={undefined}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="radio"
              value="childcare_possible_answer-103474"
            />
            No
          </label>,
        ],
        "className": "dxp-radio",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "childcare_possible_answer-103473",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className={undefined}
                disabled={false}
                name={undefined}
                onBlur={undefined}
                onChange={[Function]}
                type="radio"
                value="childcare_possible_answer-103473"
              />,
              "Yes",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": undefined,
                "disabled": false,
                "name": undefined,
                "onBlur": undefined,
                "onChange": [Function],
                "type": "radio",
                "value": "childcare_possible_answer-103473",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Yes",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "childcare_possible_answer-103474",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className={undefined}
                disabled={false}
                name={undefined}
                onBlur={undefined}
                onChange={[Function]}
                type="radio"
                value="childcare_possible_answer-103474"
              />,
              "No",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": undefined,
                "disabled": false,
                "name": undefined,
                "onBlur": undefined,
                "onChange": [Function],
                "type": "radio",
                "value": "childcare_possible_answer-103474",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "No",
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
